lib/perl/Genome/Model/Tools/DetectVariants2/Manta.pm

```diff
@@ -0,0 +1,120 @@
+package Genome::Model::Tools::DetectVariants2::Manta;
+
+use warnings;
+use strict;
+
+use Genome;
+
+use Genome::Sys::LSF::ResourceParser qw(parse_lsf_params);
+
+use File::Spec qw(join);
+use Data::Dumper;
+
+class Genome::Model::Tools::DetectVariants2::Manta {
+    is => 'Genome::Model::Tools::DetectVariants2::Detector',
+    has_param => [
+        lsf_resource => {
+            default_value => Genome::Config::get('lsf_resource_dv2_manta'),
+        },
+    ],
+};
+
+sub _gb_ram {
+    my $self = shift;
+
+    my $lsf_params = parse_lsf_params($self->lsf_resource);
+
+    my $kb_ram = $lsf_params->{'rLimits'}->{'RSS'};
+
+    if (!defined($kb_ram)) {
+        $self->fatal_message('Unable to resolve RAM from the lsf resource string: '. $self->lsf_resource);
+    }
+
+    return ($kb_ram / 1_000_000);
+}
+
+sub _detect_variants {
+    my $self = shift;
+
+    my $working_directory = $self->_temp_staging_directory;
+
+    # Optional params defined as detector params in the processing profile
+    # parse the param string to make the necessary param to hash ref translation
+    # Example : --exome --config-file=/my/foo/config.txt
+    # ( exome => 1, config_file => '/my/foo/config/txt' )
+    # Also see Genome::Model::Tools::DetectVariants2::Mutect::parse_params
+    my @resolved_config_params = (
+        '--tumor-bam-file' => $self->aligned_reads_input,
+        '--version' => $self->version,
+        '--working-directory' => $working_directory,
+        '--reference-fasta' => $self->reference_sequence_input,
+    );
+
+    # Determine which manta output VCF should be used as the "high-quality" result
+    # If single-sample, assume tumor-only (NOT GERMLINE).  We know somatic if a normal is defined.
+    my $manta_vcf_filename = 'tumorSV.vcf.gz';
+
+    if ( defined($self->control_aligned_reads_input) ) {
+        push @resolved_config_params, ('--normal-bam-file' => $self->control_aligned_reads_input);
+        $manta_vcf_filename = 'somaticSV.vcf.gz';
+    }
+
+    my @dv2_config_params = split(' ',$self->params);
```

Comment:
> How does the UR command parser get an array of values to pass to resolve_class_and_params_for_argv?

UR's "parser" relies on the shell having already split up the command line and populated ```@ARGV``` before the process starts up.  The only parsing it has to do is recognize ```--``` as the preamble to a parameter name, and know that param names can be separated from values with "=".

> Even with @known_params_list I have to parse the value or the existence of the boolean flags in the string $self->params

```@known_params_list``` is supposed to contain only property/param names, not values.  Then the map in my example will build up the right params list suitable for passing to ```create()```, whether the values are boolean or not.  Another option might be to look at UR::Object::copy().  Extract the part of it that gets the list of keys/values into a new method.  Use this new method to get the keys/values out of ```$self``` and pass them into ```Genome::Model::Tools::Manta::Config->create()```.

Maybe I'm missing something... At this point in the code, have the properties of ```$self```  been filled in yet as a result of parsing what's in ```$self->params()```?  If you really must parse this string, Text::Parsewords::shellwords() purports to break it up the same way a shell would.